### PR TITLE
[FLINK-21894][hive] Update type of HiveTablePartition#partitionSpec from Map<String, Object> to Map<String, String>

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

In order to support watermark for Hive connector (FLINK-21871), we need to extract time from the partition string values. However, currently `HiveTablePartition#partitionSpec` uses `Map<String, Object>` type to store the partition spec which is hard convert back to string values. Therefore, I propose to use the raw `Map<String, String>` type instead which is easier to convert into `Map<String, Object>` when needed.

## Brief change log

- Update type of `HiveTablePartition#partitionSpec` from `Map<String, Object>` to `Map<String, String>`

## Verifying this change

This change is a trivial rework which can be covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/JobConfWrapper.java [hunk 1]
```diff
@@ -45,6 +46,12 @@ public JobConf conf() {
         return jobConf;
     }
 
+    public String getDefaultPartitionName() {
```
Comment (lirui-apache) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/JobConfWrapper.java:49]: Can we add a util method to do this? `JobConfWrapper` is only meant to make JobConf serializable.

**Review by lirui-apache — COMMENTED** [review]
Thanks @wuchong for working on this. Just left one minor comment.

**Comment by wuchong** [comment]
Good point @lirui-apache , hope I have addressed your concern.

**Review by lirui-apache — APPROVED** [review]
+1, thanks for updating the PR

**Pull request merged**